runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -1484,11 +1484,21 @@ ClassFileOracle::walkMethodCodeAttributeCode(U_16 methodIndex)
 			}
 			break;
 
-		case CFR_BC_aload_0:
-			if (CFR_BC_getfield == code[codeIndex + 1]) {
+		case CFR_BC_aload_0: {
+			bool enableOptimization = CFR_BC_getfield == code[codeIndex + 1];
+			/* As JBaload0getfield is optimization that combines two bytecodes into one, the intermediate
+			 * state for value types are not in sync with what the stack map expects so we disable it altogether
+			 */
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			if ('Q' == methodDescriptorData[codeIndex]) {
```

Comment:
`methodDescriptorData[codeIndex]` this isnt actually correct, youre getting lucky because the aload0 is the first bytecode. The methodDesc is essentially the signature for the method, it has nothing to do with the btyecode index.

We will need to do a bigger investigation for this. Continue to do your testing with this code, but remove it from the PR once everything is ready. Remember to also run the internal value type PR build